astropy/io/ascii/tdat.py

```diff
@@ -0,0 +1,853 @@
+            if col.info.description is not None and col.info.description != "None":
+                field_line += f" // {col.info.description}"
+            elif "comment" in col_info_meta:
+                field_line += " //"
+            if "comment" in col_info_meta:
+                field_line += f" // {col_info_meta['comment']}"
+            lines.append(field_line)
+
+        if len(keywords) != 0:
+            if "parameter_defaults" in keywords:
+                lines.append("#")
+                lines.append(
+                    f"{'parameter_defaults'} = {keywords.pop('parameter_defaults')}"
+                )
+        if len(keywords) != 0:
+            lines.append("#")
+            lines.append("# Virtual Parameters")
+            lines.append("#")
+            for key in keywords:
+                lines.append(f"{key} = {keywords[key]}")
+        lines.append("#")
+        lines.append("# Data Format Specification")
+        lines.append("#")
+        lines.append(f"line[1] = {' '.join([col.info.name for col in self.cols])}")
+        lines.append("#")
+
+
+class TdatData(core.BaseData):
+    """Data Reader for TDAT format"""
+
+    comment = r"\s*(#|//)"
+    write_comment = "# "
+    splitter_class = TdatDataSplitter
+
+    def get_data_lines(self, lines):
+        """
+        READ: Override the default get_data_lines to find start and end lines.
+        """
+        # Select lines between <DATA> and <END> in file
+        try:
+            start_line = (
+                min(
+                    i
+                    for i, line in enumerate(lines)
+                    if line.strip().upper() == "<DATA>"
+                )
+                + 1
+            )
+        except ValueError:
+            raise TdatFormatError("<DATA> not found in file." + _STD_MSG)
+        try:
+            end_line = min(
+                i for i, line in enumerate(lines) if line.strip().upper() == "<END>"
+            )
+        except ValueError:
+            # <END> is an optional keyword to demarcate the end of the data.
+            # If not present the end of the document is assumed to be the end
+            # end of the data.
+            end_line = len(lines) - 1
+
+        self.data_lines = self.process_lines(lines[start_line:end_line])
+
+    def str_vals(self):
+        """WRITE: convert all values in table to a list of lists of strings.
+
+        This sets the fill values and possibly column formats from the input
+        formats={} keyword, then ends up calling table.pprint._pformat_col_iter()
+        by a circuitous path. That function does the real work of formatting.
+        Finally replace anything matching the fill_values.
+
+        Returns
+        -------
+        values : list of list of str
+        """
+        self._set_fill_values(self.cols)
+        self._set_col_formats()
+        for col in self.cols:
+            if np.issubdtype(col.dtype, np.integer):
+                if np.any(abs(col.data) > 2147483647):
```

Comment:
The allowed range for int32 is `-2**31` to `2**31 - 1`, so this test can't use `abs(...)`.